### DISCUSSION TITLE
Unable to Connect to Real-Time Streaming via Deepgram Client SDK from Specific IP Addresses

### DISCUSSION THREAD

**Discussion by akshat-Eubrics** [deepgram/community#1028] [General help] [answered]
#### Context:
We have implemented real-time streaming using the Deepgram client for our application. Recently, some of our clients have reported being unable to connect to the Deepgram WebSocket. This issue has arisen only in the past few days; previously, they were able to connect without any problems.

#### Observations:
- This issue is specific to certain clients and is consistent across all their employees.
- All employees from the affected clients are on the same Wi-Fi network, and none of them can connect.
- As developers, we are not facing this issue when testing from our own networks.

#### Questions:
1. Is there a way to confirm if certain IP addresses or IP ranges are being blocked by Deepgram's WebSocket servers?
2. If so, what steps can we take to resolve this issue (e.g., whitelisting or contacting Deepgram support)?
3. Are there any known changes to Deepgram's WebSocket server configurations in the past few days that could potentially cause this?

#### Additional Details:
- We are using the latest version of the Deepgram SDK for real-time streaming.
- No changes have been made to our application or the client networks from our end.

Any guidance or insights into how to diagnose and resolve this issue would be greatly appreciated.

Thanks in advance for your support!

**Comment by deepgram-community** [comment]
Thanks for asking your question. Please be sure to reply with as much detail as possible so the community can assist you efficiently.
<sub>Consider joining our [Discord community](https://discord.gg/deepgram) for more opportunity to engage with your fellow Deepgram users. You can earn points which can be redeemed for cool stuff by being active in our communities!</sub>

**Comment by deepgram-community** [comment]
Hey there! It looks like you haven't connected your GitHub account to your Deepgram account. You can do this at https://community.deepgram.com - being verified through this process will allow our team to help you in a much more streamlined fashion.

**Comment by deepgram-community** [comment]
It looks like we're missing some important information to help debug your issue. Would you mind providing us with the following details in a reply?

- The programming language you are working in (e.g. JavaScript, Python).
- The deepgram product you are using (e.g Speech to Text, Agent API)
- A request ID that triggered your error or issue.

**Reply by akshat-Eubrics** [reply]
- Programming Language: JavaScript
- Deepgram Product: Real-time streaming transcriptions

**Comment by lukeocodes** [comment]
Hey @akshat-Eubrics, are you logging what errors are occurring at the time? It's possible that you're hitting your concurrency limit.

**Reply by akshat-Eubrics** [reply]
Hey @lukeocodes , 
Unfortunately, we don't currently have logs to identify the specific errors, but I'll work on gathering some. In the meantime, could you let me know where I can check my concurrency limit? This would help me ensure that we're not exceeding it.

**Comment by deepgram-community** [comment, marked as answer]
You'll know if you're hitting the limit by the error codes you're getting. If your users can't connect, and in the logs you get 429 errors, that is you hitting the concurrency limit.

If your users are being disconnected, they'll be being sent an event with a reason (most likely).

There is no way right now for you to be able to proactively check your concurrency allocation, but there might be a hacky way to hit a /projects endpoint occationally and if that returns a 429, don't try to connect a user (Let me double check on that!).

You could also proxy the requests through your own backend server, which will give you better oversight on concurrency, logs, success, etc

> ![avatar](https://avatars.slack-edge.com/2024-03-28/6876915321187_66505c29ed0cfe659927_32.jpg) This message was sent by luke from Deepgram, via our community automation.
